catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroScriptActivityTest.java

```diff
@@ -0,0 +1,130 @@
+import org.catrobat.catroid.content.bricks.SetXBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.ui.BaseSettingsActivity.SETTINGS_SHOW_HINTS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.SETTINGS_SHOW_PHIRO_BRICKS;
+import static org.catrobat.catroid.uiespresso.ui.activity.utils.FormulaEditorCategoryListDataInteractionWrapper.onFormulaEditorCategory;
+import static org.catrobat.catroid.uiespresso.ui.activity.utils.ScriptListDataInteractionWrapper.onBrickCategory;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroScriptActivityTest {
+
+	private Map<String, Boolean> initialPreferences = new HashMap<>();
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject("phiroTestProject");
+		preparePhiroSettings();
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroIsFirstInBrickCategoryTest() {
+		onView(withId(R.id.button_add))
+				.perform(click());
+
+		String expectedCategoryString = UiTestUtils.getResourcesString(R.string.category_phiro);
+
+		onBrickCategory()
+				.atPosition(0)
+				.check(matches(withText(expectedCategoryString)));
+	}
+
+	@Test
+	public void phiroIsFirstInFormulaEditorDeviceCategoryTest() {
+		onView(withId(R.id.brick_set_x_edit_text))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_sensors))
+				.perform(click());
+
+		String firstPhiroSensorString = UiTestUtils.getResourcesString(R.string.formula_editor_phiro_sensor_front_left);
+
+		onFormulaEditorCategory()
+				.atPosition(0)
+				.check(matches(withText(firstPhiroSensorString)));
+	}
+
+	@After
+	public void teardown() throws Exception {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+
+		editor.putBoolean(SETTINGS_SHOW_PHIRO_BRICKS, initialPreferences.get(SETTINGS_SHOW_PHIRO_BRICKS));
+		editor.putBoolean(SETTINGS_SHOW_HINTS, initialPreferences.get(SETTINGS_SHOW_HINTS));
+		editor.apply();
+	}
+
+	private static void createProject(String projectName) {
```

Comment:
theres a createProjectAndGetStartScript in content.brick.utils.BrickTestUtils that could be used instead:
eg:
BrickTestUtils.createProjectAndGetStartScript(projectname).addBrick(new SetXBrick());